toqito/channels/tests/test_bitflip.py

```diff
@@ -0,0 +1,81 @@
+"""Test cases for the bit-flip channel."""
+
+import numpy as np
+import pytest
+
+from toqito.channels import bitflip
+
+
+def test_kraus_operators():
+    """Test if the function returns correct Kraus operators for given probability."""
+    prob = 0.3
+    kraus_ops = bitflip(prob=prob)
+    expected_k0 = np.sqrt(1 - prob) * np.eye(2)
+    expected_k1 = np.sqrt(prob) * np.array([[0, 1], [1, 0]])
+
+    np.testing.assert_almost_equal(kraus_ops[0], expected_k0)
+    np.testing.assert_almost_equal(kraus_ops[1], expected_k1)
+
+
+def test_apply_to_state_0():
+    """Test bitflip application to |0><0| state."""
+    prob = 0.3
+    rho = np.array([[1, 0], [0, 0]])  # |0><0|
+    expected_output = (1 - prob) * rho + prob * np.array([[0, 0], [0, 1]])
+    result = bitflip(rho, prob=prob)
+
+    np.testing.assert_almost_equal(result, expected_output)
+
+
+def test_apply_to_state_1():
+    """Test bitflip application to |1><1| state."""
+    prob = 0.3
+    rho = np.array([[0, 0], [0, 1]])  # |1><1|
+    expected_output = (1 - prob) * rho + prob * np.array([[1, 0], [0, 0]])
+    result = bitflip(rho, prob=prob)
+
+    np.testing.assert_almost_equal(result, expected_output)
+
+
+def test_probability_0():
+    """Test bitflip with probability 0 (no change)."""
+    rho = np.array([[0.5, 0.5], [0.5, 0.5]])
+    result = bitflip(rho, prob=0)
+    np.testing.assert_almost_equal(result, rho)
+
+
+def test_probability_1():
+    """Test bitflip with probability 1 (always flips)."""
+    rho = np.array([[0.5, 0.5], [0.5, 0.5]])
+    expected_output = np.array([[0.5, 0.5], [0.5, 0.5]])  # Same for maximally mixed state
+    result = bitflip(rho, prob=1)
+    np.testing.assert_almost_equal(result, expected_output)

```

Comment:
Same as my other comment, most of these unit tests can be condensed into 1 by using using parameterize.